beacon-chain/blockchain/currently_syncing_block.go

```diff
@@ -0,0 +1,27 @@
+package blockchain
+
+import "sync"
+
+type currentlySyncingBlock struct {
+	sync.Mutex
+	roots map[[32]byte]struct{}
+}
+
+func (b *currentlySyncingBlock) set(root [32]byte) {
+	b.Lock()
+	defer b.Unlock()
+	b.roots[root] = struct{}{}
+}
+
+func (b *currentlySyncingBlock) unset(root [32]byte) {
+	b.Lock()
+	defer b.Unlock()
+	delete(b.roots, root)
+}
+
+func (b *currentlySyncingBlock) isSyncing(root [32]byte) bool {
+	b.Lock()
```

Comment:
Not that it matters, but this can be a read lock.

Reply:
unless there's contention, I rather use simple mutexes and not pay the RW overhead.